lib/utils.bash

```diff
@@ -7,6 +7,45 @@ GREP_COLORS=
 
 ASDF_DIR=${ASDF_DIR:-''}
 ASDF_DATA_DIR=${ASDF_DATA_DIR:-''}
+BASH_MAJOR_VERSION=${BASH_VERSION%%.*}
+
+fast_basename() {
+  # if the path ends in a slash, remove it, then use parameter substitution
+  # if it doesn't, use parameter substitution on the path as-is
+  dirpath="$1"
+  [[ "$dirpath" == */ ]] && dirpath="${dirpath%/}" && dirpath="${dirpath##*/}" || dirpath="${dirpath##*/}"
+  printf "$dirpath"
```

Comment:
This, and other invocations to `printf` could have `%s` as a first parameter to prevent unintentional formatting